### PR TITLE
refactor(logging): be aware of RUST_LOG env

### PR DESCRIPTION
**Pull request by zwang28**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?
This PR supports overwriting log filters predefined in `init_risingwave_logger`, by specifying RUST_LOG environment variable.
One use case is I want to suppress certain logs in CI, to avoid large log size.

## Checklist

~~- [ ] I have written necessary rustdoc comments~~
~~- [ ] I have added necessary unit tests and integration tests~~
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

### CODE REVIEW

File: src/utils/runtime/src/lib.rs [hunk 1]
```diff
@@ -138,7 +140,7 @@ pub fn init_risingwave_logger(settings: LoggerSettings) {
         #[cfg(debug_assertions)]
         let filter = filter.with_default(Level::DEBUG);
 
-        layers.push(fmt_layer.with_filter(filter).boxed());
+        layers.push(fmt_layer.with_filter(to_env_filter(filter)).boxed());
```
Comment (BugenZhao) [inline src/utils/runtime/src/lib.rs:143]: Does this work?

```rust
let filter = filter.or(EnvFilter::from_default_env())
```

Comment (zwang28) [reply src/utils/runtime/src/lib.rs:143]: `and` works, thanks.

Comment (zwang28) [reply src/utils/runtime/src/lib.rs:143]: But it's not the same as `to_env_filter`.

- `to_env_filter` is overwriting using RUST_LOG.
- `filter.and ` cannot lower the predefined filter level in `init_risingwave_logger`.

Comment (BugenZhao) [reply src/utils/runtime/src/lib.rs:143]: I guess we can always make the default configuration stricter, and allow relaxing with env var? When `RUST_LOG` is not set, the `EnvFilter` fall back to `ERROR`. So `env_filter.or(targets_filter)` should work. 🤔

File: src/utils/runtime/src/lib.rs [hunk 2]
```diff
@@ -206,6 +208,29 @@ pub fn init_risingwave_logger(settings: LoggerSettings) {
     // TODO: add file-appender tracing subscriber in the future
 }
 
+fn to_env_filter(filter: Targets) -> EnvFilter {
```
Comment (BugenZhao) [inline src/utils/runtime/src/lib.rs:211]: Could you please add some doc to mention the "override" behavior?

**Review by BugenZhao — APPROVED** [review]
LGTM! Seems there's no other clean way to achieve this. 😰

**Review by xxchan — APPROVED** [review]

**Pull request merged**